packages/storage/src/types.ts

```diff
@@ -138,3 +138,24 @@ export interface AddItemsResponse {
   once: (type: AddItemsEventType, listener: AddItemsListener) => void;
   off: (type: AddItemsEventType, listener: AddItemsListener) => void;
 }
+
+export interface TxlSubscribeBucketEvent {
+  bucketName: string;
+  status: 'success' | 'error';
+  error?: Error;
+}
+
+export type TxlSubscribeEventData = TxlSubscribeBucketEvent;
+export type TxlSubscribeEventType = 'data' | 'error' | 'done';
+export type TxlSubscribeListener = (data: TxlSubscribeEventData) => void;
+
+export interface TxlSubscribeResponse {
+  on: (type: TxlSubscribeEventType, listener: TxlSubscribeListener) => void;
+  // DO I NEED THIS?
```

Comment:
Do you need which part?

Ideally, we should provide this `event emitter` interfaces so users can know exactly what `type` of event they can subscribe to (or else the default javascript/typescript event emitter interface would show up as `type: string`, `listener: any`.